libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs

```diff
@@ -330,78 +342,40 @@ protected virtual async Task OnRecognizerResultAsync(RecognizerResult recognizer
+        /// </summary>
+        /// <param name="application">LuisAplication object to initialize LuisReognizerOptions.</param>
+        /// <param name="options">If passed a LuisRecognizerOptionsV2 will be returned initialized with this object.</param>
+        /// <returns>LuisRecognizerOptions object.</returns>
+        private static LuisRecognizerOptions BuildLuisRecognizerOptionsV2(LuisApplication application, LuisPredictionOptions options, bool includeAPIResults)
         {
-            var luisPredictionOptions = predictionOptions == null ? _options : MergeDefaultOptionsWithProvidedOptions(_options, predictionOptions);
-
-            BotAssert.ContextNotNull(turnContext);
-
-            if (turnContext.Activity.Type != ActivityTypes.Message)
-            {
-                return null;
-            }
-
-            var utterance = turnContext.Activity?.AsMessageActivity()?.Text;
-            RecognizerResult recognizerResult;
-            LuisResult luisResult = null;
-
-            if (string.IsNullOrWhiteSpace(utterance))
-            {
-                recognizerResult = new RecognizerResult
-                {
-                    Text = utterance,
-                    Intents = new Dictionary<string, IntentScore>() { { string.Empty, new IntentScore() { Score = 1.0 } } },
-                    Entities = new JObject(),
-                };
-            }
-            else
+            var optionsV2 = options ?? new LuisPredictionOptions();
+            var luisVersionOptions = new LuisRecognizerOptionsV2(application)
             {
-                luisResult = await _runtime.Prediction.ResolveAsync(
-                    _application.ApplicationId,
-                    utterance,
-                    timezoneOffset: luisPredictionOptions.TimezoneOffset,
-                    verbose: luisPredictionOptions.IncludeAllIntents,
-                    staging: luisPredictionOptions.Staging,
-                    spellCheck: luisPredictionOptions.SpellCheck,
-                    bingSpellCheckSubscriptionKey: luisPredictionOptions.BingSpellCheckSubscriptionKey,
-                    log: luisPredictionOptions.Log ?? true,
-                    cancellationToken: cancellationToken).ConfigureAwait(false);
-
-                recognizerResult = new RecognizerResult
-                {
-                    Text = utterance,
-                    AlteredText = luisResult.AlteredQuery,
-                    Intents = LuisUtil.GetIntents(luisResult),
-                    Entities = LuisUtil.ExtractEntitiesAndMetadata(luisResult.Entities, luisResult.CompositeEntities, luisPredictionOptions.IncludeInstanceData ?? true, utterance),
-                };
-                LuisUtil.AddProperties(luisResult, recognizerResult);
-                if (_includeApiResults)
-                {
-                    recognizerResult.Properties.Add("luisResult", luisResult);
-                }
-            }
+                PredictionOptions = optionsV2,
```

Comment:
Similar comment to the one I made above, given that the PredictionOptions is going to be set, aren't the next 3 properties duplicating the data?
What happen if someone were to use such an object with inconsistent value between the 2? Which would win?

Reply:
If they can't be removed due to 3rd party having taken dependencies on them, consider having a constructor to LuisRecognizerOptionsV2 that take a LuisPredictionOptions as a parameter and initialize the properties that exist in both.

Reply:
In this case, this is a helper function to deduplicate code and force the existing obsolete constructors to use the new preferred one.